allensdk/brain_observatory/behavior/session_apis/abcs/behavior_raw_data_base.py

```diff
@@ -0,0 +1,62 @@
+import abc
+from datetime import datetime
+from typing import List
+
+
+class BehaviorRawDataBase(abc.ABC):
+    """Abstract base class implementing required methods for interacting with
+    raw (from LIMS or from JSON) behavior session data.
+    """
+
+    def get_behavior_session_id(self) -> int:
```

Comment:
what's the thought behind all these `NotImplementedError`s vs. the `abstractmethod` decorator?

Reply:
Hmmm... I guess I can slap an abstractmethod decorator on them.

Though it seems like `NotImplementedError` is okay as well:
https://docs.python.org/3/library/exceptions.html#NotImplementedError

Reply:
i should clarify. I was not requesting a change, just asking a question here.

Reply:
Ah okay, learned something important which is that `abstractmethod` will raise an error on instantiation whereas `NotImplementedError` will only raise an error on call.